addons/mail/wizard/mail_compose_message.py

```diff
@@ -195,6 +196,42 @@ def _compute_body(self):
             elif not composer.template_id or not composer.body:
                 composer.body = False
 
+    @api.depends('composition_mode', 'model', 'res_ids', 'template_id')
+    def _compute_attachment_ids(self):
+        """ Attachments computation.
+        With template
+          * mass mail: take IDs from template
+        """
+        for composer in self:
+            res_ids = composer._parse_res_ids(composer.res_ids)
+            if composer.composition_mode == 'mass_mail' and composer.template_id and composer.template_id.attachment_ids:
+                composer.attachment_ids = composer.template_id.attachment_ids
+            elif composer.composition_mode == 'comment' and len(res_ids) == 1 and composer.template_id:
+                rendered_values = composer._generate_template_for_composer(
+                    composer.template_id,
+                    res_ids,
+                    ('attachment_ids', 'attachments'),
+                )[res_ids[0]]
+                # transform attachments into attachment_ids; not attached to the
+                # document because this will be done further in the posting
+                # process, allowing to clean database if email not send
+                attachment_ids = []
+                if rendered_values.get('attachments'):
+                    attachment_ids = self.env['ir.attachment'].create([
+                        {'name': attach_fname,
+                         'datas': attach_datas,
+                         'res_model': 'mail.compose.message',
+                         'res_id': 0,
+                         'type': 'binary',    # override default_type from context, possibly meant for another model!
+                        } for attach_fname, attach_datas in rendered_values.pop('attachments')
+                    ]).ids
+                if rendered_values.get('attachment_ids'):
+                    attachment_ids += rendered_values['attachment_ids']
+                if attachment_ids:
```

Comment:
do we need this condition ? (shoudl'nt we "remove any previous attachment" if we computed no attachment here, and if we changed the template_id e.g.)